src/util/imagefiledata.cpp

```diff
@@ -0,0 +1,40 @@
+#include "util/imagefiledata.h"
+
+#include <QByteArray>
+#include <QDebug>
+#include <QFile>
+#include <QFileInfo>
+#include <QImage>
+
+// Most of the cover art extensions are 'jpg' at cover art archive.
+ImageFileData::ImageFileData(const QByteArray& coverArtBytes)
+        : QImage(QImage::fromData(coverArtBytes)), m_coverArtBytes(coverArtBytes) {
+}
+
+//static
+ImageFileData ImageFileData::fromFilePath(const QString& coverArtFilePath) {
+    QFile coverArtFile = QFile(coverArtFilePath);
+    coverArtFile.open(QIODevice::ReadOnly);
+    QByteArray coverArtBytes = coverArtFile.readAll();
+    coverArtFile.close();
+    return ImageFileData(coverArtBytes);
+}
+
+bool ImageFileData::saveFile(QString coverArtAbsoluteFilePath) const {
+    // This block added if the temp file is passed.
+    if (coverArtAbsoluteFilePath.contains('_')) {
```

Comment:
That loops unpredictable. 
The only thing we need to consider to add the right extension. 
How about: 
`ImageFileData::saveFileAddSuffix(QString pathWithoutSuffix)`

The other option would by just saveFile and assert for the correct suffix.